servant-client/src/Servant/Client/Internal/HttpClient.hs

```diff
@@ -169,9 +169,19 @@ performRequest req = do
           status_code = statusCode status
           ourResponse = clientResponseToResponse response
       unless (status_code >= 200 && status_code < 300) $
-        throwError $ FailureResponse ourResponse
+        throwError $ mkFailureResponse request ourResponse
```

Comment:
shouldn't we simply use `BaseUrl` and `Request`, then defining a new `UrlReq`?

Reply:
I suppose we could but this may end up keeping more data alive than necessary to report the error (e.g. potentially a large request body). Moreover, `ServantError` wants an `NFData` instance, which we cannot provide for `Request` without lying.